controller/engine_controller.go

```diff
@@ -1412,10 +1412,16 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 
 	replicaURL := engineapi.GetBackendReplicaURL(addr)
 	go func() {
-		defer engineClientProxy.Close()
-
 		log := ec.logger.WithField("volume", e.Spec.VolumeName)
 
+		engineClientProxyGo, err := ec.getEngineClientProxy(e, e.Status.CurrentImage)
```

Comment:
nit: `engineClientProxy` should be able to be shadowed in another scope, so basically you can use the same name as outer if you want.